localstack/utils/aws/dead_letter_queue.py

```diff
@@ -68,7 +64,51 @@ def _send_to_dead_letter_queue(source_type, source_arn, dlq_arn, event, error):
             raise Exception(msg)
     elif ':sns:' in dlq_arn:
         sns_client = aws_stack.connect_to_service('sns')
-        sns_client.publish(TopicArn=dlq_arn, Message=message, MessageAttributes=message_attrs)
+        for message in messages:
+            sns_client.publish(TopicArn=dlq_arn, Message=message['MessageBody'],
+                MessageAttributes=message['MessageAttributes'])
     else:
         LOG.warning('Unsupported dead letter queue type: %s' % dlq_arn)
     return dlq_arn
+
+
+def _prepare_messages_to_dlq(source_arn, event, error):
+    messages = []
+    custom_attrs = {
+        'RequestID': {'DataType': 'String', 'StringValue': str(uuid.uuid4())},
+        'ErrorCode': {'DataType': 'String', 'StringValue': '200'},
+        'ErrorMessage': {'DataType': 'String', 'StringValue': str(error)}
+    }
+    if ':sqs:' in source_arn:
+        custom_attrs['ErrorMessage']['StringValue'] = str(error.result)
+        for record in event['Records']:
```

Comment:
Same as below - let's better use defensive programming here, to avoid `KeyError`. 
```
for record in event.get('Records', []):
    ...
    messages.append({'Id': record.get('messageId'), 'MessageBody': record.get('body'),
                'MessageAttributes': message_attrs})
```

Reply:
Thanks for the suggestions @whummer . I'll make the modifications and push the code ASAP.